cmd/uplink/cmd/meta_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package cmd_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/internal/testrand"
+)
+
+func min(x, y int) int {
+	if x < y {
+		return x
+	}
+
+	return y
+}
+
+func TestSetGetMeta(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount:   1,
+		StorageNodeCount: 14,
+		UplinkCount:      1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		uplinkExe := ctx.Compile("storj.io/storj/cmd/uplink")
+
+		// Configure uplink.
+		{
+			output, err := exec.Command(uplinkExe,
+				"--config-dir", ctx.Dir("uplink"),
+				"setup",
+				"--non-interactive",
+				"--scope", planet.Uplinks[0].GetConfig(planet.Satellites[0]).Scope,
+			).CombinedOutput()
+			t.Log(string(output))
+			assert.NoError(t, err)
+			if err != nil {
+				return
+			}
+		}
+
+		// Create bucket.
+		bucketName := testrand.BucketName()
+
+		{
+			output, err := exec.Command(uplinkExe,
+				"--config-dir", ctx.Dir("uplink"),
+				"mb",
+				"sj://"+bucketName,
+			).CombinedOutput()
+			t.Log(string(output))
+			assert.NoError(t, err)
+			if err != nil {
+				return
+			}
+		}
+
+		// Upload file with metadata.
+		metadata := testrand.Metadata()
+
+		metadataBs, err := json.Marshal(metadata)
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+
+		metadataStr := string(metadataBs)
+
+		var metadataNorm map[string]string
+		err = json.Unmarshal(metadataBs, &metadataNorm)
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+
+		path := testrand.Path()
+		uri := "sj://" + bucketName + "/" + path
+
+		{
+			output, err := exec.Command(uplinkExe,
+				"--config-dir", ctx.Dir("uplink"),
+				"cp",
+				"--metadata", metadataStr,
+				"-", uri,
+			).CombinedOutput()
+			t.Log(string(output))
+			assert.NoError(t, err)
+			if err != nil {
+				return
+			}
+		}
+
+		// Get all metadata.
+		{
+			cmd := exec.Command(uplinkExe,
+				"--config-dir", ctx.Dir("uplink"),
+				"meta", "get", uri,
+			)
+			t.Log(cmd)
```

Comment:
I'm not sure whether it's useful to log the commands themselves.

Reply:
It was helpful when trying to figure out why the test failed. Especially for the `meta get` ones.